### PR TITLE
[IMP] im_livechat, *: do not use uuid anymore in live chat routes 

### PR DESCRIPTION
**Pull request by tsm-odoo**
*: mail, website_livechat.

Originally, the live chat controllers used the channel uuid to
authenticate requests. Since [1], the live chat relies on the discuss
guest mechanism. Several routes were updated in the process to rely on
the guest authentication instead of the channel uuid but some routes
still rely on the old uuid mechanism.

This PR replaces all the routes that still rely on the uuid mechanism
to rely on the guest one.

[1]: https://github.com/odoo/odoo/pull/129770

### CODE REVIEW

File: addons/im_livechat/controllers/cors/chatbot.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import route
+from odoo.addons.im_livechat.controllers.chatbot import LivechatChatbotScriptController
+from odoo.addons.im_livechat.tools.misc import force_guest_env
+
+
+class CorsLivechatChatbotScriptController(LivechatChatbotScriptController):
+    @route('/chatbot/cors/restart', type="json", auth="public", cors="*")
```
Comment (seb-odoo) [inline addons/im_livechat/controllers/cors/chatbot.py:9]: apply formatter on new files (this has mix of quote, and possibly other issues)

File: addons/im_livechat/controllers/main.py [hunk 2]
```diff
@@ -208,9 +208,10 @@ def _post_feedback_message(self, channel, rating, reason):
         }
         channel.message_post(body=body, message_type='notification', subtype_xmlid='mail.mt_comment')
 
-    @http.route('/im_livechat/feedback', type='json', auth='public', cors="*")
-    def feedback(self, uuid, rate, reason=None, **kwargs):
-        channel = request.env['discuss.channel'].sudo().search([('uuid', '=', uuid)], limit=1)
+    @http.route('/im_livechat/feedback', type='json', auth='public')
```
Comment (seb-odoo) [inline addons/im_livechat/controllers/main.py:211]: (double quotes on changed lines?)

File: addons/im_livechat/controllers/main.py [hunk 3]
```diff
@@ -219,13 +220,14 @@ def feedback(self, uuid, rate, reason=None, **kwargs):
                 'feedback': reason,
                 'is_internal': False,
             }
-            if not channel.rating_ids:
+            # sudo: rating.rating - visitor can access rating to give its feedback
```
Comment (seb-odoo) [inline addons/im_livechat/controllers/main.py:223]: this line in particular is not to "give feedback" but to check if it's already given (which is also ok)

File: addons/im_livechat/models/discuss_channel.py [hunk 4]
```diff
@@ -158,9 +147,10 @@ def _rating_get_parent_field_name(self):
 
     def _email_livechat_transcript(self, email):
         company = self.env.user.company_id
+        # sudo: discuss.channel - required to render the email template
```
Comment (seb-odoo) [inline addons/im_livechat/models/discuss_channel.py:150]: I wonder if appropriate sudo shouldn't be put in the template instead?
channel should be readable without sudo

File: addons/im_livechat/models/discuss_channel.py [hunk 5]
```diff
@@ -196,7 +186,8 @@ def _chatbot_find_customer_values_in_messages(self, step_type_to_field):
         """
         values = {}
         filtered_message_ids = self.chatbot_message_ids.filtered(
-            lambda m: m.script_step_id.step_type in step_type_to_field.keys()
+            # sudo: chatbot.script.step - getting the type of the current step
+            lambda m: m.script_step_id.sudo().step_type in step_type_to_field
```
Comment (seb-odoo) [inline addons/im_livechat/models/discuss_channel.py:190]: I'm surprised that the step itself is accessible

File: addons/im_livechat/models/discuss_channel.py [hunk 6]
```diff
@@ -252,11 +243,12 @@ def _message_post_after_hook(self, message, msg_vals):
         return super()._message_post_after_hook(message, msg_vals)
 
     def _chatbot_restart(self, chatbot_script):
-        self.write({
+        # sudo: discuss.channel - visitor can clear current step to restart the script
+        self.sudo().write({
             'chatbot_current_step_id': False
         })
```
Comment (seb-odoo) [inline addons/im_livechat/models/discuss_channel.py:249]: could take the opportunity to make this a single line I guess (and maybe use field = syntax ?)

File: addons/im_livechat/models/discuss_channel.py (hunk 4 above)
Comment (tsm-odoo) [reply addons/im_livechat/models/discuss_channel.py:150]: Indeed, I applied this suggestion and added the required sudo in the template instead.

File: addons/im_livechat/models/discuss_channel.py (hunk 5 above)
Comment (tsm-odoo) [reply addons/im_livechat/models/discuss_channel.py:190]: I'm surprised as well, I found out that adding `.sudo(False)` to `current_step` (chatbot.py#46) prevent from accessing the step itself. This method is only called from this flow.

File: addons/mail/controllers/discuss/rtc.py [hunk 7]
```diff
@@ -39,7 +39,7 @@ def session_call_notify(self, peer_notifications):
     @http.route("/mail/rtc/session/update_and_broadcast", methods=["POST"], type="json", auth="public")
     @add_guest_to_context
     def session_update_and_broadcast(self, session_id, values):
-        """Update a RTC session and broadcasts the changes to the members of its channel,
+        """Update a RTC session and broadcasts the changes to the members of their channel,
```
Comment (seb-odoo) [inline addons/mail/controllers/discuss/rtc.py:42]: Isn't it the channel of the rtc session in this case?

Comment (tsm-odoo) [reply addons/mail/controllers/discuss/rtc.py]: Yes it its, fixed.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Comment by seb-odoo** [comment]
@robodoo override=ci/security

`cors` routes protected by `force_guest_env`

**Pull request closed without merging**